Examples/TextEditorApp/TextEditorApp/TextEditor/TextViewModel.swift

```diff
@@ -60,47 +59,47 @@ class TextViewModel {
             // subscribe document event.
             let clientID = await self.client.id
 
-            await self.document.eventStream.sink { [weak self] event in
+            await self.document.subscribe { [weak self] event in
                 if event.type == .snapshot {
                     Task { [weak self] in
                         await self?.syncText()
                     }
                 }
-            }.store(in: &self.cancellables)
+            }
+
+            await self.document.subscribe(targetPath: "$.content") { event in
+                guard let event = event as? RemoteChangeEvent else {
+                    return
+                }
 
-            // define event handler that apply remote changes to local
-            self.textEventStream.sink { [weak self] events in
                 var textChanges = [TextOperation]()
 
-                events.filter { $0.actor != clientID }.forEach {
-                    switch $0.type {
-                    case .content:
-                        let range = NSRange(location: $0.from, length: $0.to - $0.from)
-                        let content = $0.content ?? ""
-
-                        textChanges.append(.edit(range: range, content: content))
-                    case .style:
-                        break
-                    case .selection:
-                        let range: NSRange
-
-                        if $0.from <= $0.to {
-                            range = NSRange(location: $0.from, length: $0.to - $0.from)
-                        } else {
-                            range = NSRange(location: $0.to, length: $0.from - $0.to)
-                        }
+                event.value.filter { $0.actorID != clientID }.forEach { changeInfo in
```

Comment:
Is there a need to perform the additional filtering by `actor ID` after filtering by `RemoteChangeEvent`?

Reply:
No. There was an old code left. thanks.